src/libs/dotcms.languages.js

```diff
@@ -0,0 +1,16 @@
+import DotCMSApi from './dotcms.api';
+let languagesConf = [];
+
+export const getCode = (locationSearch) => {
+    const query = new URLSearchParams(locationSearch);
+    return query.get('lang');
+};
+
+export const getId = async (langCode) => {
+    languagesConf =
+        languagesConf.length === 0
+            ? await DotCMSApi.getConfiguration().then((data) => data.languages)
```

Comment:
This is not good, everytime we ask for the lang id we hit the endpoint. We should cache the first response of the endpoint and just get it from there.

We should cache a key value map like:

```js
const langs = {
    'en': 1,
    'es': 2,
    'fr': 3
...
}
```

Reply:
no, the endpoint is being requested only once, then it's stored in `languagesConf`, so the next time a request is done to `getId()` languagesConf already has the map of languages